docs/adr/367-branch-protection-md

```diff
@@ -0,0 +1,109 @@
+
+- Risk of direct or accidental pushes to protected branches
+- Increased likelihood of unreviewed or unsafe code merges
+- Reduced ability to enforce standardized development workflows
+
+GitHub allows fine-grained control over branch protections, including the ability for administrators (tc members, repo captains...) to bypass rules when needed (e.g., urgent hotfixes or security patches).
+
+This decision aims to normalize a secure-by-default stance while maintaining flexibility for urgent, responsible exceptions.
+
+## Decision
+
+**Branch protection rules will be enabled by default on all repositories.**
+
+Key rules to be applied:
+
+- Protect release branches, this includes: `main`, `master`, and `v*` branches/tags
+- Require pull request reviews before merging (at least 1 approval)
+- Require all status checks to pass before merging
+
+Optional rules:
+- Enforce linear history (disable merge commits, allow squash/rebase)
+- Restrict who can push directly to the protected branches to people who can release (captains, TC, etc)
+
+### Exceptions:
+
+- TC members and Repo Captains will retain the ability to **temporarily bypass branch protections** in critical scenarios:
+  - Security fixes that require immediate deployment
+  - Emergency production incidents
+  - When CI is clearly broken for unrelated reasons
+  - To fix history when necessary
+  - When a reasonable case is made where the exception makes sense
+- These exceptions must be documented in the relevant PR or release communication
+
+## Rationale
+
+### Alternatives Considered:
+
+- **Opt-in branch protection**: Rejected due to inconsistent adoption and higher risk of accidental pushes.
+- **No admin bypass**: Rejected because it adds overhead in urgent situations and may block time-sensitive fixes.
+
+### Pros and Cons:
+
+**Pros:**
+
+- Stronger default security posture
+- Better alignment with compliance and audit requirements
+- Reduced chance of unsafe or accidental changes
+
+**Cons:**
+
+- May slow down development for repos lacking automation (e.g., CI failures delaying merges)
+- Risk of misuse of admin bypass unless monitored
+
+**Why is this the best option?**
+
+This decision strikes a balance between **security and agility**, using defaults that promote safe practices while preserving escape hatches for legitimate exceptions.
+
+## Consequences
+
+### Positive Impact:
+
+- Improved consistency and safety across all codebases
+- Reduced likelihood of accidental or unreviewed code merges
+- Easier to audit and enforce development standards across org
+
+### Negative Impact:
+
+- Initial friction for projects with a single captain or no committers (at least 1 approval per PR)
+- Possible misuse of admin bypass if not documented
+
+
+## Implementation
+
+### Phase 1: Rollout to Active Repos
+
+- Identify all active repositories
+- Notify repo maintainers of the change
+- Apply default protection rules using GitHub API or organization policies
```

Comment:
<img width="1914" height="456" alt="CleanShot 2025-07-13 at 11 20 40@2x" src="https://github.com/user-attachments/assets/6459f255-7112-4a6a-a8da-d22ca8f32ac8" />

Can you document the API you use so people can re-use/update it over time? 🙏